### DISCUSSION TITLE
Theory behind local thickness filter

### DISCUSSION THREAD

**Discussion by javierpagalo** [PMEAL/porespy#494] [Q&A] [answered]
Hello, 
I am looking for theory for local thickness method. As seen in the code it is porosimetry approach but with no shielding effect.
Reviewing the researches of Gostick et al. in "Insights into the Effect of Structural Heterogeneity in Carbonized Electrospun Fibrous Mats for Flow Battery Electrodes by X-Ray Tomography" and "In situ Compression and X-ray Computed Tomography of Flow Battery Electrodes", they refers that porespy implementation of Local Thickness procedure is based on M. Y. M. Chiang et al, in "Local Thickness and Anisotropy Approaches to Characterize Pore Size Distribution of Three-Dimensional Porous Networks".

Is that the actual theory behind the Local Thickness Porespy function?
or Is it other study behind that?

Thank you
Waiting for your response

**Comment by jgostick** [comment, marked as answer]
The "theory" is just that you put the largest spheres you can in the image.  How you do it is less important...that only affects speed and memory usage for instance.  The paper we refer to in our previous work uses skeletons which are not always robust since they don't always go where they're supposed to.  Our method is robust in that we iteratively find all locations where spheres of size r=R can find, then r=R-1, then r=R-2, etc.  On each iteration we update the image with "r" that have not already been updated on a previous iteration.  I do not know of a paper that explicitly outlines our method, but as I said it's just a background implementation detail and the result is the same.

**Reply by javierpagalo** [reply]
Ok, It helps me to understand. 
Thanks a lot.
